src/crc32c.c

```diff
@@ -116,8 +117,8 @@ static uint32_t crc32c_sse42(uint32_t crc, const char *buf, size_t len)
             __asm__(CRC32_PTR "\t" "(%3), %0\n\t"
                     CRC32_PTR "\t" LONGx1 "(%3), %1\n\t"
                     CRC32_PTR "\t" LONGx2 "(%3), %2"
-                    : "=r"(crc0), "=r"(crc1), "=r"(crc2)
-                    : "r"(buf), "0"(crc0), "1"(crc1), "2"(crc2));
+                    : "+r"(crc0), "+r"(crc1), "+r"(crc2)
+                    : "r"(buf), "m"(*buf));
```

Comment:
I merged this, but reverted it.  Because the `buf` pointer is incremented throughout the loop, it is not necessarily a pointer to at least `LONG*3` bytes.

(Besides, in each iteration aren't only 4 bytes being read/written?)

Reply:
`*buf` only reads 1 byte, so it would be incorrect for the assembly to read more than that. This assembly reads `LONG*3` bytes starting at pointer however

Reply:
Oh, I see, they are not sequential reads though, but rather `LONG` bytes apart, so we need to write out the indexing call for each of those:
```suggestion
                    : "r"(buf),
                    "m"(* (const char (*)[sizeof(void*)]) &buf[0]),
                    "m"(* (const char (*)[sizeof(void*)]) &buf[LONG]),
                    "m"(* (const char (*)[sizeof(void*)]) &buf[LONG*2]));
```